mu/virtual_environment.py

```diff
@@ -383,6 +417,42 @@ def ensure_interpreter(self):
             logger.error(message)
             raise VirtualEnvironmentError(message)
 
+    def ensure_interpreter_version(self):
+        """Ensure that the venv interpreter matches the version of Python running Mu
+
+        This is necessary because otherwise we'll have mismatched wheels etc.
+        """
+        current_version = "%s%s" % sys.version_info[:2]
+        #
+        # Can't use self.run_python as we're not yet within the Qt UI loop
+        #
+        process = subprocess.run([self.interpreter, "-c", 'import sys; print("%s%s" % sys.version_info[:2])'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True)
+        venv_version = process.stdout.decode("utf-8").strip()
+        if current_version == venv_version:
+            logger.info("Both interpreters at version %s", current_version)
+        else:
+            message = (
+                "Current interpreter is at version %s; venv interpreter is at version %s" % (current_version, venv_version)
+            )
+            logger.error(message)
+            raise VirtualEnvironmentError(message)
+
+    def ensure_key_modules(self):
+        """Ensure that the venv interpreter is able to load key modules
+        """
+        #
+        # FIXME: import from wheels.mode_packages, but need import name, not PyPI name
+        #
+        modules = ["pygame", "pgzero", "flask", "serial"]

```

Comment:
Should we specify that the first entry of the `mode_packages` tuples needs to be the "import name"?

Reply:
Thing is, we also need the PyPI name for the pip download/install. We could have both, but it didn't seem worth complicating it further. This is just a sanity check.

Reply:
But the PyPI name (or argument for `pip install`) is included in the second entry of the tuple:
https://github.com/mu-editor/mu/blob/69e430e613fa7c2fb6743e9d9f9ae9b4edab7eaf/mu/wheels/__init__.py#L18
https://github.com/mu-editor/mu/blob/69e430e613fa7c2fb6743e9d9f9ae9b4edab7eaf/mu/wheels/__init__.py#L14-L17

So the first entry is only used for printing info, no? Am I missing its usage from somewhere else?

Reply:
Sorry; you're quite right. I was moving too fast yesterday to check. In that case, yes, I'll change the "name" field to be the importable name.